rules/library.bzl

```diff
@@ -971,7 +971,8 @@ def apple_library(name, library_tools = {}, export_private_headers = True, names
             "@build_bazel_rules_ios//:virtualize_frameworks": framework_vfs_objc_copts,
             "//conditions:default": framework_vfs_objc_copts if enable_framework_vfs else [],
         })
-
+    if module_map:
```

Comment:
We solve a problem with modulemaps by appending the module_map to the objc headers list. An alternative approach is here: https://github.com/bazel-ios/rules_ios/commit/7f4fe87a13edd72c9276bfbdc2d41155c464e571

However this approach may be preferable because bazel sees that modulemap is a dependency of the objc-library target.

Reply:
For this change, we return module_map target from the apple_library macro, since the objc-provider in bazel5 no longer supports it. See bazelbuild/bazel@31bec27#diff-eb9c179576423931e9db781a5acff548a82a4361a23ce1c1dc0c6b73b30d114bL35 and slack conversation here: https://ios-dev-at-scale.slack.com/archives/CBVEC7KRQ/p1637262740176000